src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -201,6 +231,23 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ * Requests diagnostic logs of the desired types from the device.
+ *
+ * @param type       The type of logs being requested.
+ * @param timeout    The timeout for getting the logs.  If the timeout expires, completion
+ *                   will be called with whatever logs have been retrieved by that point
+ *                   (which might be none, and might include partial logs).
+ *                   If the timeout is null, logs will be retreived when processing is
+ *                   complete and completion is called.
+ * @param queue      The queue on which completion will be called.
+ * @param completion The completion that will be called to pass in the file paths for the requested logs.
+ */
+- (void)getDiagnosticLogsOfTypes:(MTRDiagnosticLogType)type
+                         timeout:(NSInteger * _Nullable)timeout
```

Comment:
Wait, why are we using NSInteger here?  This part does not make sense.

Reply:
so we can have a null timeout here. i used a NSTimeInterval before as per our conversation but you said we can't support nullable using NSTimeInterval. see

Reply:
done.